src/Modules/CalcOffence.lua

```diff
@@ -1788,55 +1816,63 @@ function calcs.offence(env, actor, activeSkill)
 					end
 					t_insert(breakdown.Speed, s_format("= %.2f ^8(casts per second)", output.Repeats / output.Cooldown))
 					t_insert(breakdown.Speed, s_format("\n"))
-					t_insert(breakdown.Speed, s_format("= %.2f ^8(lower of cast rates)", output.Speed))
+					t_insert(breakdown.Speed, s_format("= %.2f ^8(lower of cast rates)", output.DisplaySpeed))
+				end
+				if  output.TempUptimeVariable < 1 then
+					t_insert(breakdown.Speed, s_format("x %.2f ^8(offensive uptime)",  output.TempUptimeVariable))
 				end
 			end
 			if breakdown and calcLib.mod(skillModList, skillCfg, "SkillAttackTime") > 0 then
 				breakdown.Time = { }
 				breakdown.multiChain(breakdown.Time, {
-					base = { "%.2f ^8(base)", 1 / (output.Speed * calcLib.mod(skillModList, skillCfg, "SkillAttackTime") ) },
+					base = { "%.2f ^8(base)", 1 / (output.DisplaySpeed * calcLib.mod(skillModList, skillCfg, "SkillAttackTime") ) },
 					{ "%.2f ^8(total modifier)", calcLib.mod(skillModList, skillCfg, "SkillAttackTime")  },
-					total = s_format("= %.2f ^8seconds per attack", output.Time)
+					total = s_format("= %.2f ^8seconds per attack", output.DisplayTime)
 				})
 			end 
 		end
 		if skillData.hitTimeOverride and not skillData.triggeredOnDeath then
-			output.HitTime = skillData.hitTimeOverride
+			output.HitTime = skillData.hitTimeOverride *  output.TempUptimeVariable
 			output.HitSpeed = 1 / output.HitTime
+			output.DisplayHitSpeed = 1 / skillData.hitTimeOverride
 			--Brands always have hitTimeOverride
 			if skillCfg.skillName and skillCfg.skillName:match("Brand") then
 				output.BrandTicks = m_floor(output.Duration * output.HitSpeed)
 			end
 		elseif skillData.hitTimeMultiplier and output.Time and not skillData.triggeredOnDeath then
-			output.HitTime = output.Time * skillData.hitTimeMultiplier
+			output.HitTime = output.Time * skillData.hitTimeMultiplier *  output.TempUptimeVariable
 			output.HitSpeed = 1 / output.HitTime
+			output.DisplayHitSpeed = 1 / output.Time * skillData.hitTimeMultiplier
 		end
 	end
 	if breakdown then
 		breakdown.SustainableTrauma = storedSustainedTraumaBreakdown
 	end
 	output.SustainableTrauma = activeSkill.activeEffect.grantedEffect.name == "Boneshatter" and skillModList:Sum("BASE", skillCfg, "Multiplier:SustainableTraumaStacks")
-
+	output.TempUptimeVariable =  tempUptimeVariable
 	if isAttack then
 		-- Combine hit chance and attack speed
 		combineStat("AccuracyHitChance", "AVERAGE")
 		combineStat("HitChance", "AVERAGE")
 		combineStat("Speed", "AVERAGE")
 		combineStat("HitSpeed", "OR")
+		combineStat("DisplaySpeed", "AVERAGE")
+		combineStat("DisplayHitSpeed", "OR")
 		if output.Speed == 0 then
 			output.Time = 0
 		else
+			output.DisplayTime = 1 / output.Speed *  output.TempUptimeVariable
```

Comment:
```suggestion
			output.DisplayTime = 1 / (output.Speed *  output.TempUptimeVariable)
```
You want 50% of speed not 50% of time. Switching to division also works here if you don't want the brackets.

Reply:
Not quite, but it was still wrong. That section sets the time for MH+OH combined speed. The Speed stat already has the uptime variable factored into it during the passes. The intent of that section was to remove the variable again from the result.

That said, that's better accomplished by setting DisplayTime = 1 / DisplaySpeed, since that's the same variable as Speed but without the uptime variable.